### PR TITLE
Add versiontool

### PR DESCRIPTION
**Pull request by rynowak**
This tool can update versions of projects and dependencies across a whole
repo at the commandline.

### CODE REVIEW

File: src/VersionTool/Program.cs [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            {
+                paths.Add(Path.GetFullPath("."));
+            }
+
+            foreach (var project in EnumerateProjects(paths))
+            {
+                Console.WriteLine($"Name:      {project.Name}");
+                Console.WriteLine($"Directory: {project.ProjectDirectory}");
+                Console.WriteLine($"Version:   {project.Version}");
+                Console.WriteLine();
```
Comment (javiercn) [inline src/VersionTool/Program.cs]: Nit: You can use our logging infrastructure instead of doing console writeline. You get nice colors, verbosity, etc.

**Comment by Eilon** [comment]
@pranavkm all yours.

File: src/VersionTool/Program.cs (hunk 1 above)
Comment (rynowak) [inline src/VersionTool/Program.cs]: won't fix, this is specifically formatted text output for a console, not generic logging.

File: src/VersionTool/Program.cs [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+        private static int OnListDependency(
+            CommandOption pathOption,
+            CommandOption matchingOption,
+            CommandArgument dependencyArgument)
+        {
+            var paths = pathOption.Values;
+            if (paths.Count == 0)
+            {
+                paths.Add(Path.GetFullPath("."));
```
Comment (pranavkm) [inline src/VersionTool/Program.cs]: `Directory.GetCurrentDirectory()`

File: src/VersionTool/Program.cs [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+            return 0;
+        }
+
+        private static IEnumerable<Project> EnumerateProjects(IEnumerable<string> paths)
+        {
+            return paths.SelectMany(p =>
+            {
+                return
+                    Directory.EnumerateFiles(p, "project.json", SearchOption.AllDirectories)
```
Comment (pranavkm) [inline src/VersionTool/Program.cs]: Could we limit this to `src`, `test`, `samples` and `tools`? I'm not sure if repos have random directories with things that we shouldn't be touching.

File: src/VersionTool/Program.cs [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                if (frameworkDependencies != null)
+                {
+                    foreach (var dependency in frameworkDependencies.Properties())
+                    {
+                        yield return dependency;
+                    }
+                }
+            }
+
+            var runtimes = root.Property("runtimes")?.Value as JObject;
```
Comment (pranavkm) [inline src/VersionTool/Program.cs]: o.o, didn't know this worked. Good to know

File: src/VersionTool/Program.cs [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+            using (var stream = File.OpenRead(path))
+            {
+                var reader = new JsonTextReader(new StreamReader(stream));
+                return JObject.Load(reader);
+            }
+        }
+
+        private static void SaveJObject(string path, JObject root)
+        {
+            using (var stream = File.Open(path, FileMode.Truncate, FileAccess.Write))
```
Comment (pranavkm) [inline src/VersionTool/Program.cs]: Just make sure this preserves the BOM.

File: src/VersionTool/project.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "Microsoft.DotNet.ProjectModel": "1.0.0-*",
+    "Microsoft.NETCore.App": {
+      "type": "platform",
+      "version": "1.0.0"
+    }
+  },
+
+  "frameworks": {
+    "netcoreapp1.0": {
+      "imports": ["dnxcore50"]
```
Comment (pranavkm) [inline src/VersionTool/project.json]: Remove

**Comment by pranavkm** [comment]
:shipit:

File: src/VersionTool/Program.cs (hunk 5 above)
Comment (rynowak) [inline src/VersionTool/Program.cs]: It did when I ran it a few weeks ago... good enough?

File: src/VersionTool/Program.cs (hunk 3 above)
Comment (rynowak) [inline src/VersionTool/Program.cs]: It really doesn't seem like a good idea to put a limitation in a command line tool based on our repo structure. If we want something like this you'd just call the tool in a loop.

File: src/VersionTool/Program.cs (hunk 2 above)
Comment (rynowak) [inline src/VersionTool/Program.cs]: :+1:

**Pull request closed without merging**